service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh

```diff
@@ -756,6 +756,34 @@ fi
 chown ${LOGIN_USER}:${LOGIN_USER} "${USER_BASHRC}"
 chown ${LOGIN_USER}:${LOGIN_USER} "${USER_BASH_PROFILE}"
 
+#######################################
+# Restart proxy to pick up the new env
+######################################
+readonly APP_PROXY=$(get_metadata_value "instance/attributes/terra-app-proxy")
+readonly PROXY_ENV="/opt/deeplearning/proxy.env"
+if [[ -n "${APP_PROXY}" ]]; then
+    emit "Using custom Proxy Agent"
+    RESOURCE_ID=$(get_metadata_value "instance/attributes/terra-resource-id")
+    NEW_PROXY="https://${APP_PROXY}"
+    NEW_PROXY_URL="${RESOURCE_ID}.${APP_PROXY}"
+    # escape all the orrurences of forward slash
+    ESCAPED_NEW_PROXY=$(sed 's/[\/&]/\\&/g' <<< "${NEW_PROXY}")
+    sed -i "s/^PROXY_REGISTRATION_URL=.*/PROXY_REGISTRATION_URL=${ESCAPED_NEW_PROXY}/" "${PROXY_ENV}"
```

Comment:
Sed allows you to use any separation character (not just forward slash), so you can eliminate the escaping of the URL and just do:

```
    NEW_PROXY="https://${APP_PROXY}"
    NEW_PROXY_URL="${RESOURCE_ID}.${APP_PROXY}"

    # Update the proxy.env file with new URLs and backend ID
    sed -i "s#^PROXY_REGISTRATION_URL=.*#PROXY_REGISTRATION_URL=${ESCAPED_NEW_PROXY}#" "${PROXY_ENV}"
    sed -i "s#^PROXY_URL=.*#PROXY_URL=${NEW_PROXY_URL}#" "${PROXY_ENV}"
    sed -i "s#^BACKEND_ID=.*#BACKEND_ID=${RESOURCE_ID}#" "${PROXY_ENV}"

    # With the proxy.env updated, restart the notebooks-proxy-agent
    systemctl restart notebooks-proxy-agent.service

    emit "Proxy Agent service restarted"
```